lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,25 @@
+module PulpProxy
+  class DiskUsage
+    attr_accessor :free, :total, :block_device, :path
+
+    def initialize(path = PulpProxy::Plugin.settings.pulp_dir)
+      @path = path
+    end
+
+    def has_df_h?
+      df_h.nil? ? false : true
+    end
+
+    def to_json
+      return {}.to_json unless has_df_h?
+      disk_data = df_h.split
+      {:path => @path, :free => disk_data[3], :total => disk_data[1], :mounted_on => disk_data.last}.to_json
+    end
+
+    private
+
+    def df_h
+      `df -h #{@path}`.split("\n").last
```

Comment:
if you use which, then you wont be callign df multiple times, if you decide to keep calling it, at least cache the resuts

also, imho you should check for df error codes etc.